src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncSpillTests.cs

```diff
@@ -3888,5 +3887,2493 @@ class AltBoolean
             CompileAndVerify(source, expectedOutput: "True", options: TestOptions.DebugExe);
             CompileAndVerify(source, expectedOutput: "True", options: TestOptions.ReleaseExe);
         }
+
+        [Fact]
+        [WorkItem(42755, "https://github.com/dotnet/roslyn/issues/42755")]
+        public void KeepLtrSemantics_ClassFieldAccessOnProperty()
+        {
+            var source = @"
+using System;
+using System.Threading.Tasks;
+
+class Program
+{
+    static async Task Main(string[] args)
+    {
+        var a = new A();
+        try
+        {
+            a.Invalid.x = await Write(""Invalid"");
+        }
+        catch 
+        {
+            a.Valid.x = await Write(""Valid"");
+        }                       
+                                      
+        async Task<int> Write(string s)
+        {
+            await Task.Yield();
+            Console.WriteLine(s);
+            return 5;
+        }
+    }
+}
+
+class A
+{
+    public B Valid => new B();
+    public B Invalid => throw new Exception();
+}
+
+class B
+{
+    public int x;
+}";
+            var comp = CSharpTestBase.CreateCompilation(source, options: TestOptions.ReleaseExe);
+            CompileAndVerify(comp, expectedOutput: "Valid")
```

Comment:
>Valid [](start = 52, length = 5)

I think that we would want to make it clear from the output what exception is thrown and at what point it is thrown, I suggest we print types of the exception caught and print something distinct before each throwing statement in order to make it clear where the exception was thrown.
Also, I think we would want to test behavior of scenarios when exceptions aren't thrown at all. Preferably within the same unit-test. This can be achieved by calling the same function with different inputs.